src/event/log_event.rs

```diff
@@ -92,6 +93,48 @@ impl Into<BTreeMap<String, Value>> for LogEvent {
     }
 }
 
+impl From<HashMap<String, Value>> for LogEvent {
+    fn from(map: HashMap<String, Value>) -> Self {
+        LogEvent {
+            fields: map.into_iter().collect(),
+        }
+    }
+}
+
+impl Into<HashMap<String, Value>> for LogEvent {
+    fn into(self) -> HashMap<String, Value> {
+        let Self { fields } = self;
+        fields.into_iter().collect()
```

Comment:
Out of curiosity, (how) does this differ from `self.fields.into_iter().collect()`?

Reply:
In this case we destructure the `Self` first, so we own the `fields`, as opposed to borrowing them.